src/main/java/nextstep/qna/domain/Answers.java

```diff
@@ -0,0 +1,80 @@
+package nextstep.qna.domain;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import nextstep.qna.CannotDeleteException;
+import nextstep.users.domain.NsUser;
+
+public class Answers {
+    
+    private final List<Answer> answers;
+    
+    public Answers() {
+        this(List.of());
+    }
+    
+    public Answers(Answer... answers) {
+        this(Arrays.asList(answers));
+    }
+    
+    public Answers(List<Answer> answers) {
+        this.answers = new ArrayList<>(answers);
+    }
+    
+    public void addAnswer(Answer answer) {
+        this.answers.add(answer);
+    }
+    
+    public List<DeleteHistory> deleteAll(NsUser loginUser) throws CannotDeleteException {
+        isHaveAuthority(loginUser);
+        arsAnswersDeleteDone();
```

Comment:
```suggestion
        for(Answer answer: this.answers) {
            answer.delete(loginUser);
        }
```

answer에 대해서도 같은 방식으로 구현해야하지 않을까?
지금과 같이 구현하는 것은 answer를 수동적인 객체로 바라보는 느낌이 듦

Reply:
answers 를 일급컬렉션으로 만드는것에 집중한 나머지 answer 을 수동적으로 바라보아
삭제 권한 검증을 answers 에서 하게되고, 그와 별도로 삭제 자체도 answers 에서 시도하게 되었습니다.

answer 에서 필요한이유는
1. answers 에 검증과 삭제가 있으면 answer 은 getter만 남는 자기 행동이 없는 수동적인 객체가 되는 점
2. 또한 answers 가 answer 의 삭제 로직을 대신 수행하게 되는 점
때문입니다.